index.js

```diff
@@ -343,6 +343,10 @@
     return xBool !== yBool ? or(x, y) : xEmpty;
   });
 
+  var wrapIndexOf = function(f) {
+    return R.curry(R.compose(R.ifElse(R.lte(0), Just, Nothing), f));
```

Comment:
I agree that the placeholder avoids misinterpretation of a partially applied `R.{lt,gt}{,e}`.

> Actually would there be a way to make the whole function definition point free?

Yesterday, between talks at LambdaConf, I did just that, but I didn't have a reliable internet connection so didn't post it.

```javascript
//  compose :: (b -> c) -> (a -> b) -> a -> c
var compose = R.curryN(2, R.compose);

var sanctifyIndexOf = R.pipe(compose(Just),
                             compose(filter(R.gte(_, 0))),
                             R.curry);
```

This is not necessarily as good as your version of the function, but it was an enjoyable exercise.

I do like the name `sanctifyIndexOf`, though. :)